### ISSUE TITLE
OR queries don't work anymore

### ISSUE THREAD

**Issue by yenda** [realm/realm-js#2398]
## Goals

The is no IN queries in realm-js https://github.com/realm/realm-js/issues/450

The recommended workaround in to use OR queries:
`var filtered = sample.filtered([2,4,7,10].map((id) => 'id == ' + id).join(' OR '));`

But they are broken now in 2.28 (2.26 was fine)

## Expected Results

```
for query:
message-id=="0x3bf387af09cfbbce6d100aa601dfca2ca67918899e99eb358619aa227a3472f3"
result length:
1
```

```
for query:
message-id=="0x3bf387af09cfbbce6d100aa601dfca2ca67918899e99eb358619aa227a3472f3" OR message-id=="0x4aa5bb0b3b9644fc033fd330fda07ccded34334624eff703b4cb9414df4c5047"
result length:
2
```

## Actual Results

```
for query:
message-id=="0x3bf387af09cfbbce6d100aa601dfca2ca67918899e99eb358619aa227a3472f3"
result length:
1
```

```
for query:
message-id=="0x3bf387af09cfbbce6d100aa601dfca2ca67918899e99eb358619aa227a3472f3" OR message-id=="0x4aa5bb0b3b9644fc033fd330fda07ccded34334624eff703b4cb9414df4c5047"
result length:
0
```
## Steps to Reproduce

In the repl I tried:
```
status-im.data-store.user-statuses> (-> @core/account-realm
        (.objects "user-status")
        (.filtered "( message-id == \"0x3bf387af09cfbbce6d100aa601dfca2ca67918899e99eb358619aa227a3472f3\" OR message-id==\"0xb3abc6823d7f2890e29d04537df1a6941342aa64ceb865b5386fc600ba942bad\")")
        .-length)
0
status-im.data-store.user-statuses> (-> @core/account-realm
        (.objects "user-status")
        (.filtered "( message-id == \"0x3bf387af09cfbbce6d100aa601dfca2ca67918899e99eb358619aa227a3472f3\")")
        .-length)
1
```

## Code Sample

https://github.com/status-im/status-react

## Version of Realm and Tooling

- Realm JS SDK Version: 2.28.0
- Node or React Native: react native 0.59.3
- Client OS & Version: Android
- Which debugger for React Native: None

**Comment by bmunkholm** [comment]
I would be very surprised if OR queries don't work as we have unit tests for that - but we could be wrong of course. For the code sample, could you please provide a self-contained minimal sample that shows the error? That would be highly appreciated - thanks!

**Comment by kneth** [comment]
@yenda I haven't been able to reproduce it. Please take a look at #2542 to see if I have tested the wrong thing.

**Comment by ironage** [comment]
@kneth This looks like another case of string-or queries not working when the search string is more than 64 characters long. It is recently fixed by https://github.com/realm/realm-core/pull/3386 and available in core v5.23.5.

**Comment by kneth** [comment]
So the fix has been released in JS v2.3.0.
